constraintlayout/core/src/main/java/androidx/constraintlayout/core/state/helpers/HorizontalChainReference.java

```diff
@@ -42,6 +42,12 @@ public void apply() {
                     first.startToStart(mStartToStart).margin(mMarginStart);
                 } else if (mStartToEnd != null) {
                     first.startToEnd(mStartToEnd).margin(mMarginStart);
+                } else if (mLeftToLeft != null) {
+                    // TODO: Hack until we support RTL properly
+                    first.startToStart(mLeftToLeft).margin(mMarginLeft);
+                } else if (mLeftToRight != null) {
+                    // TODO: Hack until we support RTL properly
+                    first.startToEnd(mLeftToRight).margin(mMarginLeft);
```

Comment:
This is the weirdest bit, PTAL.

It's necessary since the constraints are resolved for RTL in compose, but they are exclusively handled with start/end in the chain.